tools/assets-automation/assets-maintenance-tool/Azure.Sdk.Tools.Assets.MaintenanceTool/Scan/AssetsScanner.cs

```diff
@@ -0,0 +1,357 @@
+
+                if (results != null)
+                {
+                    return new AssetsResultSet(results);
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Given a repo configuration, scan the repo and return the assetsresults from all targeted branches.
+        /// </summary>
+        /// <param name="config"></param>
+        /// <param name="previousOutput"></param>
+        /// <returns></returns>
+        private List<AssetsResult> ScanRepo(RepoConfiguration config, AssetsResultSet? previousOutput)
+        {
+            string? envOverride = Environment.GetEnvironmentVariable(GIT_TOKEN_ENV_VAR);
+            var authString = string.Empty;
+            if (!string.IsNullOrWhiteSpace(envOverride))
+            {
+                authString = $"{envOverride}@";
+            }
+
+            var targetRepoUri = $"https://{authString}github.com/{config.Repo}.git";
+            var workingDirectory = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+            var results = new List<AssetsResult>();
+
+            try
+            {
+                if(!Directory.Exists(workingDirectory))
+                {
+                    Directory.CreateDirectory(workingDirectory);
+                }
+
+                foreach(var branch in config.Branches)
+                {
+                    var commitsOnBranch = GetBranchCommits(targetRepoUri, branch, config.ScanStartDate, workingDirectory);
+                    var unretrievedCommits = ResolveUnhandledCommits(commitsOnBranch, previousOutput);
+
+                    results.AddRange(GetAssetsResults(config.Repo, unretrievedCommits, workingDirectory));
+
+                    if (previousOutput != null)
+                    {
+                        foreach (var commit in commitsOnBranch.Where(x => !unretrievedCommits.Contains(x)))
+                        {
+                            results.AddRange(previousOutput.ByOriginSHA[commit]);
+                        }
+                    }
+                }
+            }
+            finally
+            {
+                CleanupWorkingDirectory(workingDirectory);
+            }
+
+            return results;
+        }
+
+        /// <summary>
+        /// Clones a specific branch, then returns all commit shas newer than our targeted date.
+        /// </summary>
+        /// <param name="uri"></param>
+        /// <param name="branch"></param>
+        /// <param name="workingDirectory"></param>
+        /// <returns></returns>
+        private List<string> GetBranchCommits(string uri, string branch, DateTime since, string workingDirectory)
+        {
+            var commitSHAs = new List<string>();
+            try
+            {
+                // if git is already initialized, we just need to checkout a specific branch
+                if (!Directory.Exists(Path.Combine(workingDirectory, ".git"))) {
```

Comment:
I see two conventions for bracketing here. `if` has `{` on the same line but `else` has on `{` on newline. Let's be consistent everywhere?